tests/workflows/test_pytorch_optuna.py

```diff
@@ -0,0 +1,208 @@
+# Derived partially from https://pytorch.org/tutorials/beginner/dcgan_faces_tutorial.html
+import numpy as np
+import optuna
+import pytest
+import torch
+import torch.nn as nn
+import torch.nn.parallel
+import torch.optim as optim
+import torch.utils.data
+from dask.distributed import PipInstall, wait
+
+NC = 3  # Number of channels in the training images. For color images this is 3
+NZ = 100  # Size of z latent vector (i.e. size of generator input)
+
+
+# We create a fake image dataset. This ought to be replaced by
+# your actual dataset or pytorch's example datasets. We do this here
+# to focus more on computation than actual convergence.
+class FakeImageDataset(torch.utils.data.Dataset):
```

Comment:
Do you have any suggestions on a real dataset to use here?

Reply:
Yesh... could. But I approached this workflow with the objective of making the most use of the GPUs as possible and not complicating it with downloading external data (which my first attempt revield being hit with rate limits, so could have to copy to our S3 I suppose)

In general though, the layout follows the standard PyTorch pattern of Dataset/DataLoader so anyone looking at it as reference would have a short way to go to get their dataset.

With that said, the image dataset isn't entirely random, in that it sorta moves the values of the image to the value of the label; but I yield to your judgement, if it's better to have real data I'm happy to adjust.

Reply:
I'll push back here.  While I appreciate having a demo during conferences that has high utilization (thanks for that, it was useful), I also appreciate having workflows that represent real workloads.  

If, for example, we find that it's very hard to saturate a GPU due to data loading then as an engineering team we should focus much more on data loading.  If we find that in real world use cases data loading is not important then we can ignore it comfortably.  

Having a realistic workload with realistic data loading helps us, I think, to answer questions like these.

These workflows serve multiple use cases.  For notebooks I think demo-ability is useful.  For the pytest benchmarks I think that fidelity to the real-world is probably more useful.

Reply:
I agree with Matt here. We want real data for this and if necessary we'll copy to our S3

Reply:
Let it be written then, I'll update. 👍

Reply:
Updated in https://github.com/coiled/benchmarks/pull/787/commits/ca8b0fd7805fe3e6df44289a11c2de3f7f569318